### PR TITLE
[auth-swift] Swift implementation of link anon to email password

### PR DESCRIPTION
**Pull request by paulb777**
Swift implementation of #12009

There are no AuthTests.swift changes because some tests were missed in #11942. I'll follow up with those in a separate PR.

### CODE REVIEW

File: FirebaseAuth/Sources/Swift/Backend/RPC/SignUpNewUserRequest.swift [hunk 1]
```diff
@@ -77,6 +82,12 @@ class SignUpNewUserRequest: IdentityToolkitRequest, AuthRPCRequest {
       @brief The password inputed by the user.
    */
   private(set) var displayName: String?
+
+  /** @property idToken
+      @brief he idToken of the user.
```
Comment (andrewheard) [inline FirebaseAuth/Sources/Swift/Backend/RPC/SignUpNewUserRequest.swift:87]: typo: `he` instead of `The`

**Review by andrewheard — APPROVED** [review]

File: FirebaseAuth/Sources/Swift/Backend/RPC/SignUpNewUserRequest.swift (hunk 1 above)
Comment (paulb777) [reply FirebaseAuth/Sources/Swift/Backend/RPC/SignUpNewUserRequest.swift]: Thanks. Looks like I shouldn't include the dash in the commit message to skip CI https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

File: FirebaseAuth/Sources/Swift/MultiFactor/Phone/PhoneMultiFactorInfo.swift [hunk 2]
```diff
@@ -55,10 +55,10 @@ import Foundation
     override public class var supportsSecureCoding: Bool { return secureCodingWorkaround }
 
     public required init?(coder: NSCoder) {
-      guard let phoneNumber = coder.decodeObject(forKey: kPhoneNumberCodingKey) as? String else {
+      guard let phoneNumber = coder.decodeObject(forKey: kPhoneNumberCodingKey) as? NSString else {
```
Comment (renkelvin) [inline FirebaseAuth/Sources/Swift/MultiFactor/Phone/PhoneMultiFactorInfo.swift:58]: Is `NSString` available in Swift?

Comment (paulb777) [reply FirebaseAuth/Sources/Swift/MultiFactor/Phone/PhoneMultiFactorInfo.swift:58]: Yes. It addresses a runtime warning with NSCoder. It still gets converted to String when assigned to phoneNumber.

Comment (renkelvin) [reply FirebaseAuth/Sources/Swift/MultiFactor/Phone/PhoneMultiFactorInfo.swift:58]: Good to know, thanks.

**Review by renkelvin — APPROVED** [review]

**Pull request merged**